### ISSUE TITLE
Potential invalid offence with Lint/EndAlignment on case statements.

### ISSUE THREAD

**Issue by savef** [bbatsov/rubocop#2317]
I'm getting a Lint/EndAlignment offence with the following code:
```ruby
process case value
when 1 then 'one'
when 2 then 'two'
end
```

I would expect that using `AlignWith: variable` with this cop it would treat method calls as just as valid as variable assignments. Supposing you don't agree with that there should be another switch for this cop to make it not handle case statements.

I'm using v0.34.2. I think this offence will only have been present since PR #2157. Note that I also use `Style/CaseIndentation: IndentWhenRelativeTo: end`.

Thanks.

**Comment by lumeet** [comment]
I agree, the cop shouldn't complain about this. PR submitted.

**Comment by savef** [comment]
Nice, thanks!

**Comment by jonas054** [comment]
What about
```ruby
return case value
when 1 then 'one'
when 2 then 'two'
end
```
Should that also be included in `AlignWith: variable`? Or would it be better to introduce `AlignWith: start_of_line` that enforces alignment with the beginning of the line where the keyword is?

**Comment by lumeet** [comment]
I guess this is trickier than I first thought. I don't really know what to do with even harder scenarios...

```ruby
return process case value
when 1 then 'one'
end,
case other
when 2 then 'two'
end
```

A new style with such a subtle difference sounds a bit of an overkill to me, too. One possible solution is to ignore these cases altogether by `Lint/EndAlignment`... and perhaps introduce a new cop with an option to discourage the use of `case` this way. :)
